File: schedule_app/lib/models/response.dart
import 'package:enum_to_string/enum_to_string.dart';
import 'package:schedule_app/enumators/alert.dart';
import 'package:schedule_app/enumators/response_type.dart';

class Response<T> {
  String message;
  ResponseType tipo;
  bool hasError;
  T data;
  Alert alert;

  Response();

  Response.messageError(String message) {
    this.message = message;
    this.hasError = true;
  }

  Response.fromJson(Map<String, dynamic> json) {
    message = json['message'];
    tipo = EnumToString.fromString(ResponseType.values, json['tipo']);
    hasError = json['hasError'];
  }
}

File: schedule/src/main/java/com/metting/scheduling/schedule/utils/Response.java
package com.metting.scheduling.schedule.utils;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Response<T> {

    private T data;
    private String message;
    private ResponseType tipo;
    private Boolean hasError = false;
    private Alert alert;

    public Response() {

    }
    public Response(T obj) {
        this.data = obj;
        this.tipo = ResponseType.SUCCESS;
    }

    public Response(String message, ResponseType type) {
        this.message = message;
        this.tipo = type;
        this.hasError = true;
    }
}
